### PR TITLE
Adding avatar to the crews

### PR DESCRIPTION
**Pull request by Minilik94**

### CODE REVIEW

File: apps/web/src/lib/components/ui/community-details/libraryDetails.svelte
Comment (antopiahk) [inline apps/web/src/lib/components/ui/community-details/libraryDetails.svelte:1]: Naming convention is in discussion in the discord, but at least use the same standard within the same folder (the other files are "PascalCase". so it should be "LibraryDetails.svelte".

but in the shadcn format, they should all be "kebab-case", so make them all kebab case as they are in the shadcn folder format.
agent-details.svelte
crew-details.svelte
library-details.svelte

File: apps/web/src/routes/app/sessions/Message.svelte
Comment (antopiahk) [inline apps/web/src/routes/app/sessions/Message.svelte:1]: good job not leaving anything outdated here! 🥳 
But next time, if the file is not related to your branch or if you know someone is working on that file (I was since I am working on the sessions feature), make sure you let them know if it affects them!

File: apps/web/src/lib/components/ui/library/crew-library.svelte [hunk 1]
```diff
@@ -9,78 +9,76 @@
...
 	let showDetailInTheModal = async (id: string) => {
 		displayedAgent = myCrews.find((a) => a.id === id) || publishedCrews.find((a) => a.id === id);
-		console.log(displayedAgent);
 	};
 
-
 	function handleClose() {
 		showDetails = false;
 	}
+
```
Comment (antopiahk) [inline apps/web/src/lib/components/ui/library/crew-library.svelte]: @nazifbara is there a better way to do this, fundamentally? It doesn't seem too optimal 🤔 
I think creating component(s) instead is "better". what do you think?
crew-library.svelte also has similar ``reusableClasses`` variable.

File: apps/web/src/lib/components/ui/community-details/AgentDetails.svelte [hunk 2]
```diff
@@ -0,0 +1,68 @@
+<script lang="ts">
+	import dayjs from 'dayjs';
+	import relativeTime from 'dayjs/plugin/relativeTime';
+	import type { Agent } from '$lib/types/models';
+
+	dayjs.extend(relativeTime);
+
+	export let AgentDisplayDetails: Agent;
```
Comment (antopiahk) [inline apps/web/src/lib/components/ui/community-details/AgentDetails.svelte]: naming convention 👀 should be "camelCase"

File: apps/web/src/lib/components/ui/community-details/AgentDetails.svelte [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+
+	function timeSince(dateIsoString: Date | string | number) {
```
Comment (antopiahk) [inline apps/web/src/lib/components/ui/community-details/AgentDetails.svelte]: if I'm not mistaken, this is the new version instead of the many if statements, great stuff!! 💪 
Would love to have this in utils.ts though, so I can use this function as well :)

File: apps/web/src/lib/components/ui/community-details/AgentDetails.svelte [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+		return dayjs(dateIsoString).fromNow(true);
+	}
+</script>
+
+<div class="mb-8 flex flex-col items-center justify-center">
+	<div class="relative">
+		<img
+			src={AgentDisplayDetails?.avatar}
```
Comment (antopiahk) [inline apps/web/src/lib/components/ui/community-details/AgentDetails.svelte]: Take some time to learn about `null` and `undefined` types in TypeScript, very important for good code!
This article might help: https://www.w3schools.com/typescript/typescript_null.php
Understand what the `?.` is used for and how to use it properly. 
If help needed on this, let me know! 👍

File: apps/web/src/lib/components/ui/community-details/CrewDetails.svelte [hunk 5]
```diff
@@ -0,0 +1,67 @@
+<script lang="ts">
+	import dayjs from 'dayjs';
+	import relativeTime from 'dayjs/plugin/relativeTime';
+	import type { Crew } from '$lib/types/models';
+
+	dayjs.extend(relativeTime);
+
+	export let CrewDisplayDetails: Crew;
```
Comment (antopiahk) [inline apps/web/src/lib/components/ui/community-details/CrewDetails.svelte]: just another naming nitpick

File: apps/web/src/lib/components/ui/community-details/libraryDetails.svelte
Comment (antopiahk) [inline apps/web/src/lib/components/ui/community-details/libraryDetails.svelte:1]: It seems like you're using components a little incorrectly. Some general issues. like file names aren't too clear. types are incorrect. "library-details" is quite misleading name. Should be something that explains what it does while being simple. In this case, its literally **just** the library. the word "details" shouldn't need to be added.
Also for the ``agent-details`` and ``crew-details``, a better name instead of "details" is "item" or "row"

File: apps/web/src/lib/components/ui/community-details/libraryDetails.svelte [hunk 6]
```diff
@@ -1,31 +1,21 @@
...
-	import dayjs from 'dayjs';
-	import relativeTime from 'dayjs/plugin/relativeTime';
 	import type { Agent, Crew } from '$lib/types/models';
-
-	dayjs.extend(relativeTime);
+	import { AgentDetails, CrewDetails } from '$lib/components/ui/community-details';
 
 	export let type: 'agent' | 'crew';
-	export let displayedAgent: Agent | Crew;
+	export let displayedAgent: Agent | Crew
```
Comment (antopiahk) [inline apps/web/src/lib/components/ui/community-details/libraryDetails.svelte]: if this can be a crew, the variable name shouldn't be "agent".
use a generic name like "displayedItem" that can be both. Or use two separate variables.

File: apps/web/src/lib/components/ui/community-details/libraryDetails.svelte [hunk 7]
```diff
@@ -37,109 +27,9 @@
 			: 'relativ bg-background max-w-5xl transform rounded-xl p-8 shadow-xl transition-all duration-500 ease-in-out hover:scale-105'}
```
Comment (antopiahk) [inline apps/web/src/lib/components/ui/community-details/libraryDetails.svelte]: typo with `relativ` not sure if it's breaking anything or not

**Review by antopiahk — CHANGES_REQUESTED** [review]

File: apps/web/src/lib/components/ui/community-details/libraryDetails.svelte [hunk 8]
```diff
@@ -1,31 +1,21 @@
...
 		dispatch('close');
 		showDetails = false;
-		console.log(showDetails);
-	}
-
-	function timeSince(dateIsoString: Date | string | number) {
-		return dayjs(dateIsoString).fromNow(true);
 	}
 
 	$: isAgent = type === 'agent';
```
Comment (antopiahk) [inline apps/web/src/lib/components/ui/community-details/libraryDetails.svelte]: all this logic here seems unnecessary since you're already checking the type in the component below. (line 29)

**Review by antopiahk — CHANGES_REQUESTED** [review]

File: apps/web/src/lib/components/ui/community-details/AgentDetails.svelte (hunk 3 above)
Comment (Minilik94) [reply apps/web/src/lib/components/ui/community-details/AgentDetails.svelte]: Yeah, nazif told me about dayjs library , the function name inside utils.ts is the same timeSince

File: apps/web/src/lib/components/ui/community-details/AgentDetails.svelte (hunk 4 above)
Comment (Minilik94) [reply apps/web/src/lib/components/ui/community-details/AgentDetails.svelte]: okay i will , i thought the agentDisplay might return undefined that why i used ?.

File: apps/web/src/lib/components/ui/library/crew-library.svelte (hunk 1 above)
Comment (Minilik94) [reply apps/web/src/lib/components/ui/library/crew-library.svelte]: removed the reusableClasses and created a component(agent-row and crew-row.svelte)

**Review by antopiahk — APPROVED** [review]
